arc

```diff
@@ -0,0 +1,113 @@
+#!/bin/bash
+
+set -e
+
+usage () {
+  echo "Usage: $0 [OPTIONS] <INPUT-FILE> [-- <MLIR-OPTS>]"
+  echo "Options:"
+  echo "  -h       Show this help message"
+  echo "  -o <DIR> Output directory (defaults to ./target/)"
+  exit 0
+}
+
+while getopts ho: opt; do
+    case $opt in
+        h)
+            usage
+            ;;
+        o)
+            CRATE_DIR=$OPTARG
+            ;;
+        \?)
+            echo "Invalid option: -$OPTARG" >&2
+            usage
+            exit 1
+            ;;
+    esac
+done
+
+while [[ $# -gt 0 ]]; do
+    case $1 in
+        --)
+            shift
+            break
+            ;;
+        *)
+            INPUT_FILE=$1
+            ;;
+    esac
+    shift
+done
+
+CRATE_NAME=$(basename $INPUT_FILE)-crate
```

Comment:
Quote the RHS or else things will break when `$INPUT_FILE` contains a path with a space in it.

`CRATE_NAME="$(basename "$INPUT_FILE")-crate"`

This applies to all variables in the script.

Should the crate name contain the extension of the input file?